cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -118,12 +107,31 @@ public synchronized ApiResponse registerSchema(ClusterSchemaRequest clusterSchem
       return ApiResponse.builder().result("Failure in registering schema.").build();
     } finally {
       // Ensure the Schema compatibility is returned to previous setting before the force update.
-      // (Set isForce to false in all cases to revert it.
-      if (clusterSchemaRequest.isForceRegister() && schemaCompatibilityCompleted) {
-        log.info(
-            "RegisterSchema - Force Commit revert to original Schema Compatibility {} for Topic {}",
-            schemaCompatibility,
-            clusterSchemaRequest.getTopicName());
+      resetCompatibilityOnSubject(
+          clusterSchemaRequest, schemaCompatibility, schemaCompatibilitySetOnSubject);
+    }
+  }
+
+  private void resetCompatibilityOnSubject(
+      ClusterSchemaRequest clusterSchemaRequest,
+      String schemaCompatibility,
+      boolean subjectSchemaCompatibilitySet) {
+    if (clusterSchemaRequest.isForceRegister()) {
+      if (!subjectSchemaCompatibilitySet) // check if subject compatibility is not set before
+      {
+        // get global compatibility on subject level
+        schemaCompatibility =
+            getGlobalSchemaCompatibility(
+                clusterSchemaRequest.getEnv(),
+                clusterSchemaRequest.getProtocol(),
+                clusterSchemaRequest.getClusterIdentification());
+      }
+      log.info(
+          "RegisterSchema - Force Commit revert to original Schema Compatibility {} for Topic {}",
+          schemaCompatibility,
+          clusterSchemaRequest.getTopicName());
```

Comment:
This is not true. In case `schemaCompatibility` is null or `NOT_SET` it will never perform the action, so it won't revert anything it will leave it as it is.

Reply:
Moved inside the block